### PR TITLE
State variable under contract's name

### PR DESCRIPTION
**Pull request by pirapira**
An attempt at #988.  I'm opening a pull-request just to see test results.

### CODE REVIEW

**Comment by pirapira** [comment]
message for @pirapira next Monday: I still need to add more end-to-end tests from the comments in #988.

**Comment by pirapira** [comment]
message for @pirapira monday: also add tests that use libraries.

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -5666,6 +5666,87 @@ BOOST_AUTO_TEST_CASE(accessor_for_const_state_variable)
...
+	char const* sourceCode = R"(
+		contract A {
+			uint x = 7;
+		}
+		contract B {
+			uint x = 9;
+		}
+		contract C is A, B {
+			function f() returns (uint) {
+				return A.x;
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp:5758]: Can you please add a second function that returns `B.x`? And also a function that does `B.x = 10` and another `A.x=12`?

**Review by chriseth — CHANGES_REQUESTED** [review]

**Comment by pirapira** [comment]
@chriseth this one is ready.

File: libsolidity/analysis/TypeChecker.cpp [hunk 2]
```diff
@@ -1376,6 +1376,11 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	}
 	else if (exprType->category() == Type::Category::FixedBytes)
 		annotation.isLValue = false;
+	else if (TypeType const* typeType = dynamic_cast<decltype(typeType)>(exprType.get()))
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp:1379]: oh wow, learned something new :-)

**Review by chriseth — APPROVED** [review]

**Pull request merged**